### PR TITLE
display preset options at landscape mode

### LINKED ISSUE #10379: Show the preset configuration options at the landscape screen
Add a select box to the landscape screen with the preset configurations. The modules should be automatically selected after choosing one of the preset configurations.

See https://github.com/jhipster/jhipster-lite/issues/9491 description for the complete roadmap.

### To-Do
- [x] Render `LandscapePresetConfigurationVue` component
- [x] Select modules from selected preset
- [x] Retain module selection state when preset option is deselected
- [x] Deselect preset option when a new module is selected
- [x] Retain preset option when trying to select a new non-selectable module
- [x] Deselect preset option when a selected module is deselected

### PR DESCRIPTION
**Pull request by renanfranca**
Fix #10379

there is a to-do at [this issue](https://github.com/jhipster/jhipster-lite/issues/10379) description.

### CODE REVIEW

**Comment by DamnClin** [comment]
Good feature but I think this is taking too much space and will be annoying for people on small screens. We should probably do something like the minimap for this feature (and probably make minimap and preset smaller with just an icon)

**Comment by renanfranca** [comment]
> Good feature but I think this is taking too much space and will be annoying for people on small screens. We should probably do something like the minimap for this feature (and probably make minimap and preset smaller with just an icon)

@DamnClin: Please let me know if it is okay now. I attached a new demo video below. Thank you for taking a look.

https://github.com/user-attachments/assets/90bbf1a8-7728-48d3-b741-4c18f54d2d56

**Comment by pascalgrimaud** [comment]
Is it ready for being merged? There are probably other small improvements but it can be done in another PRs

File: src/main/style/atom/select/_select.scss [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  }
+
+  &.-small {
+    padding: $jhlite-small-select-padding;
+  }
+
+  &.-rounded-top {
+    border-radius: 0;
+    border-top-left-radius: $jhlite-select-radius;
+    border-top-right-radius: $jhlite-select-radius;
```
Comment (qmonmert) [inline src/main/style/atom/select/_select.scss:49]:
```suggestion
    border-radius: $jhlite-select-radius $jhlite-select-radius 0 0;
```

File: src/main/webapp/app/module/secondary/RestPresetModuleToApply.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { ModuleSlug } from '../domain/ModuleSlug';
+
+export interface RestPresetModuleToApply {
+  slug: string;
+}
+
+export const mapPresetModulesToApply = (modules: RestPresetModuleToApply[]): ModuleSlug[] => {
+  return modules.map(module => new ModuleSlug(module.slug));
+};
```
Comment (qmonmert) [inline src/main/webapp/app/module/secondary/RestPresetModuleToApply.ts:9]:
```suggestion
export const mapPresetModulesToApply = (modules: RestPresetModuleToApply[]): ModuleSlug[] =>
  modules.map(module => new ModuleSlug(module.slug));
```

File: src/test/webapp/unit/module/primary/landscape/LandscapePresetConfigurationComponent.spec.ts [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    const options = wrapper.findAll('option');
+    expect(options.length - 1).toBe(defaultPresets().presets.length);
+    options
+      .filter((option, index) => index > 0)
+      .forEach((option, index) => {
+        expect(option.text()).toBe(defaultPresets().presets[index].name);
+      });
+  });
+
+  it('should handle API error gracefully', async () => {
```
Comment (qmonmert) [inline src/test/webapp/unit/module/primary/landscape/LandscapePresetConfigurationComponent.spec.ts:46]: `gracefully` 🧐🙂

Comment (renanfranca) [reply src/test/webapp/unit/module/primary/landscape/LandscapePresetConfigurationComponent.spec.ts:46]: I will remove the gracefully 😆, it was too much 😅

**Comment by renanfranca** [comment]
Thank you very much for taking the time to review @qmonmert ! I appreciated it 💙😃

File: src/test/webapp/unit/module/primary/landscape/LandscapeComponent.spec.ts [hunk 4]
```diff
@@ -1100,6 +1102,90 @@ describe('Landscape', () => {
...
+      const presetComponent = wrapper.findComponent(LandscapePresetConfigurationVue);
+
+      return { wrapper, presetComponent };
+    };
+
+    const selectPresetOption = async (presetComponent: VueWrapper<InstanceType<typeof LandscapePresetConfigurationVue>>, value: string) => {
+      const presetDropdown = presetComponent.find('select');
+      await presetDropdown.setValue(value);
+      await presetDropdown.trigger('change');
+    };
```
Comment (qmonmert) [inline src/test/webapp/unit/module/primary/landscape/LandscapeComponent.spec.ts:1187]:
```suggestion
    const selectPresetOption = (presetComponent: VueWrapper<InstanceType<typeof LandscapePresetConfigurationVue>>, value: string) => {
      const presetDropdown = presetComponent.find('select');
      presetDropdown.setValue(value);
      presetDropdown.trigger('change');
    };
```

File: src/test/webapp/unit/module/primary/landscape/LandscapeComponent.spec.ts [hunk 5]
```diff
@@ -1100,6 +1102,90 @@ describe('Landscape', () => {
...
+      const classes = initModuleElement.classes();
+      ['-selected', '-compacted'].forEach(expectedClass => {
+        expect(classes).toContain(expectedClass);
+      });
+    });
+
+    it('should retain module selection state when preset option is deselected', async () => {
+      const { presetComponent, initModuleElement } = await setupAndSelectPreset('init-maven');
+
+      await selectPresetOption(presetComponent, '');
```
Comment (qmonmert) [inline src/test/webapp/unit/module/primary/landscape/LandscapeComponent.spec.ts:1125]:
```suggestion
      selectPresetOption(presetComponent, '');
```

File: src/test/webapp/unit/module/primary/landscape/LandscapeComponent.spec.ts [hunk 6]
```diff
@@ -1100,6 +1102,90 @@ describe('Landscape', () => {
...
+      await flushPromises();
+
+      const presetDropdown = presetComponent.find('select');
+      expect(presetDropdown.element.value).toBe('');
+    });
+
+    const setupAndSelectPreset = async (presetValue: string) => {
+      const { wrapper, presetComponent } = await setupPresetTest();
+
+      await selectPresetOption(presetComponent, presetValue);
```
Comment (qmonmert) [inline src/test/webapp/unit/module/primary/landscape/LandscapeComponent.spec.ts:1169]:
```suggestion
      selectPresetOption(presetComponent, presetValue);
```

**Comment by qmonmert** [comment]
> Thank you very much for taking the time to review @qmonmert ! I appreciated it 💙😃

👍

**Review by qmonmert — APPROVED** [review]

**Pull request merged**